torch/distributed/nn/_functions.py

```diff
@@ -0,0 +1,339 @@
+import torch
+from torch.autograd import Function
+import torch.distributed as dist
+
+
+def broadcast(tensor, src, group=dist.group.WORLD):
+    """
+    Broadcasts the tensor to the whole group.
+
+    ``tensor`` must have the same number of elements in all processes
+    participating in the collective.
+
+    Arguments:
+        tensor (Tensor): Data to be sent if ``src`` is the rank of current
+            process, and tensor to be used to save received data otherwise.
+        src (int): Source rank.
+        group (ProcessGroup, optional): The process group to work on
+
+    Returns:
+        Tensor: received tensor from the broadcast op.
+
+    """
+    return _Broadcast.apply(src, group, tensor)
+
+
+def broadcast_multigpu(tensors, src, group=dist.group.WORLD, src_tensor=0):
+    """
+    Broadcasts the tensor to the whole group with multiple GPU tensors
+    per node.
+
+    ``tensor`` must have the same number of elements in all the GPUs from
+    all processes participating in the collective. each tensor in the list must
+    be on a different GPU
+
+    Only nccl and gloo backend are currently supported
+    tensors should only be GPU tensors
+
+    Arguments:
+        tensor_list (List[Tensor]): Tensors that participate in the collective
+            operation. If ``src`` is the rank, then the specified ``src_tensor``
+            element of ``tensor_list`` (``tensor_list[src_tensor]``) will be
+            broadcast to all other tensors (on different GPUs) in the src process
+            and all tensors in ``tensor_list`` of other non-src processes.
+            You also need to make sure that ``len(tensor_list)`` is the same
+            for all the distributed processes calling this function.
+
+        src (int): Source rank.
+        group (ProcessGroup, optional): The process group to work on
+        src_tensor (int, optional): Source tensor rank within ``tensor_list``
+
+    Returns:
+        list[Tensor]: received tensors from the broadcast op
```

Comment:
I think these are tuples (same for the other functions here).

Reply:
good catch!